### PR TITLE
Amqp gearama2

### PR DESCRIPTION
**Pull request by gearama**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

**Review by LarryOsterman — APPROVED** [review]

File: sdk/eventhubs/azure-messaging-eventhubs/vcpkg/Config.cmake.in [hunk 1]
```diff
@@ -4,7 +4,8 @@
 @PACKAGE_INIT@
 
 include(CMakeFindDependencyMacro)
-find_dependency(azure-core-cpp "1.8.0")
+find_dependency(azure-core-cpp "1.9.0")
```
Comment (antkmsft) [inline sdk/eventhubs/azure-messaging-eventhubs/vcpkg/Config.cmake.in:7]:
```suggestion
find_dependency(azure-core-cpp "1.9.0")
find_dependency(azure-core-amqp-cpp)
```

File: sdk/eventhubs/azure-messaging-eventhubs/CMakeLists.txt [hunk 2]
```diff
@@ -33,37 +32,86 @@ include(AzureBuildTargetForCI)
...
+        message("add folder ${oneFolder}")
+        add_subdirectory(${oneFolder} EXCLUDE_FROM_ALL)
+    endforeach()
+elseif(NOT AZ_ALL_LIBRARIES)
+  find_package(azure-core-cpp "1.9.0" CONFIG QUIET)
+  find_package(azure-storage-blobs-cpp "12.7.0" CONFIG QUIET)
+  if(NOT azure-core-cpp_FOUND)
+    find_package(azure-core-cpp "1.9.0" REQUIRED)
+    find_package(azure-storage-blobs-cpp "12.7.0" REQUIRED)
+  endif()
```
Comment (antkmsft) [inline sdk/eventhubs/azure-messaging-eventhubs/CMakeLists.txt:47]:
```suggestion
  find_package(azure-core-cpp "1.9.0" CONFIG QUIET)
  if(NOT azure-core-cpp_FOUND)
    find_package(azure-core-cpp "1.9.0" REQUIRED)
  endif()
  
  find_package(azure-core-amqp-cpp CONFIG QUIET)
  if(NOT azure-core-amqp-cpp_FOUND)
    find_package(azure-core-amqp-cpp REQUIRED)
  endif()
  
  find_package(azure-storage-blobs-cpp "12.7.0" CONFIG QUIET)
  if(NOT azure-core-cpp_FOUND)
    find_package(azure-storage-blobs-cpp "12.7.0" REQUIRED)
  endif()
```

File: sdk/eventhubs/azure-messaging-eventhubs/CMakeLists.txt [hunk 3]
```diff
@@ -33,37 +32,86 @@ include(AzureBuildTargetForCI)
...
       $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/inc>
       $<INSTALL_INTERFACE:include>
 )
+# Include shared source code
+# NOTE: Use shared-code only within .cpp files. DO NEVER consume the shared-code from header files.
+target_include_directories(
+  azure-messaging-eventhubs
+    PRIVATE
+      $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../azure-messaging-eventhubs/inc>
+)
```
Comment (antkmsft) [inline sdk/eventhubs/azure-messaging-eventhubs/CMakeLists.txt:103]:
```suggestion
```

File: sdk/eventhubs/azure-messaging-eventhubs/CMakeLists.txt [hunk 4]
```diff
@@ -75,11 +123,13 @@ endif()
...
   )
 
 az_rtti_setup(
   azure-messaging-eventhubs
   MESSAGING_EVENTHUBS
-  "azure/messaging/eventhubs/rtti.hpp"
+  azure/messaging/eventhubs/rtti.hpp
 )
+
+unset(FETCH_SOURCE_DEPS CACHE)
```
Comment (antkmsft) [inline sdk/eventhubs/azure-messaging-eventhubs/CMakeLists.txt:135]:
```suggestion
unset(FETCH_SOURCE_DEPS CACHE)

```

File: cmake-modules/FolderList.cmake [hunk 5]
```diff
@@ -30,6 +30,10 @@ macro(GetFolderList project)
     elseif(${project} STREQUAL STORAGE_QUEUES)
         DownloadDepVersion(sdk/core azure-core 1.5.0)
         DownloadDepVersion(sdk/storage/azure-storage-common azure-storage-common 12.2.3)
+    elseif(${project} STREQUAL MESSAGING)
+        DownloadDepVersion(sdk/core azure-core 1.8.0)
+        DownloadDepVersion(sdk/storage/azure-storage-common azure-storage-common 12.3.1)
```
Comment (antkmsft) [inline cmake-modules/FolderList.cmake:35]:
```suggestion
    elseif(${project} STREQUAL MESSAGING)
        DownloadDepVersion(sdk/core azure-core 1.8.0)
        DownloadDepVersion(sdk/core azure-core-amqp)
        DownloadDepVersion(sdk/storage/azure-storage-common azure-storage-common 12.3.1)
```

File: CMakeSettings.json [hunk 6]
```diff
@@ -56,7 +56,8 @@
           "value": "True",
           "type": "BOOL"
         }
-      ]
+      ],
+      "cmakeToolchain": "D:/vcpkg/scripts/buildsystems/vcpkg.cmake"
```
Comment (RickWinter) [inline CMakeSettings.json:60]: Don't hardcode the path.

File: sdk/eventhubs/azure-messaging-eventhubs/inc/azure/messaging/eventhubs/models/amqp_message.hpp [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+      Body.SetMessageBody(message);
+
+      message.ApplicationProperties = std::move(ApplicationProperties);
+      message.Properties.ContentType = std::move(ContentType);
+      message.Properties.CorrelationId = std::move(CorrelationId);
+
+      return message;
+    };
+  };
+}}}} // namespace Azure::Messaging::EventHubs::Models
```
Comment (RickWinter) [inline sdk/eventhubs/azure-messaging-eventhubs/inc/azure/messaging/eventhubs/models/amqp_message.hpp:98]:
```suggestion
}}}} // namespace Azure::Messaging::EventHubs::Models

```

**Review by LarryOsterman — CHANGES_REQUESTED** [review]
Taking a break from the PR now, but I have a bunch of stuff here :).

File: sdk/eventhubs/azure-messaging-eventhubs/inc/azure/messaging/eventhubs/consumer_client.hpp [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+
+    /**@brief  GetPartitionProperties gets properties for a specific partition. This includes data
+     * like the last enqueued sequence number, the first sequence number and when an event was last
+     * enqueued to the partition.
+     *
+     * @param partitionID partition ID to detail.
+     * @param options Additional options for getting partition properties
+     */
+    Models::EventHubPartitionProperties GetPartitionProperties(
+        std::string const& partitionID,
```
Comment (antkmsft) [inline sdk/eventhubs/azure-messaging-eventhubs/inc/azure/messaging/eventhubs/consumer_client.hpp:154]:
```suggestion
        std::string const& partitionId,
```

File: sdk/eventhubs/azure-messaging-eventhubs/inc/azure/messaging/eventhubs/event_data_batch.hpp [hunk 9]
```diff
@@ -0,0 +1,212 @@
...
+  /**@brief EventDataBatch is used to efficiently pack up EventData before sending it to Event Hubs.
+   *
+   * @remark EventDataBatch's are not meant to be created directly. Use
+   * [ProducerClient.NewEventDataBatch], which will create them with the proper size limit for your
+   * Event Hub.
+   */
+  class EventDataBatch {
+  private:
+    std::mutex m_rwMutex;
+    std::string m_partitionID;
```
Comment (antkmsft) [inline sdk/eventhubs/azure-messaging-eventhubs/inc/azure/messaging/eventhubs/event_data_batch.hpp:19]:
```suggestion
    std::string m_partitionId;
```

File: sdk/eventhubs/azure-messaging-eventhubs/inc/azure/messaging/eventhubs/event_data_batch.hpp [hunk 10]
```diff
@@ -0,0 +1,212 @@
...
+      {
+        SetMaxBytes(std::numeric_limits<uint16_t>::max());
+      }
+      else
+      {
+        SetMaxBytes(options.MaxBytes);
+      }
+    };
+
+    void SetPartitionID(std::string partitionID) { m_partitionID = partitionID; }
```
Comment (antkmsft) [inline sdk/eventhubs/azure-messaging-eventhubs/inc/azure/messaging/eventhubs/event_data_batch.hpp:87]:
```suggestion
    void SetPartitionId(std::string const& partitionId) { m_partitionID = partitionID; }
```

**Review by antkmsft — COMMENTED** [review]
I have not gone through every occurrence, but search the code for "ID" case sensitive, they should be changed to "Id".
ID - is not an abbreviation, and we use "Id" in all other places in the SDK.

**Pull request closed without merging**